### ISSUE TITLE
pug under gulp

### ISSUE THREAD

**Issue by Scorpovi4** [pugjs/pug#2591]
How could I modify configuration of pug under gulp to exclude closing watch mode after every compile error?

```
gulp.task('pug', function(){
    return gulp.src('app/pug/**/*.pug')
    .pipe(pug({pretty: true}))
    .pipe(gulp.dest('app'))
});
```

**Comment by PascaleBeier** [comment]
Do you make use of the [gulp.watch](https://github.com/gulpjs/gulp/blob/master/docs/API.md#gulpwatchglob-opts-tasks) API?

```javascript
gulp.watch('app/pug/**/*.pug', ['pug']);
```

**Comment by Scorpovi4** [comment]
Yes, I use gulp.watch API and it stop during compilation if I did mistake into *.pug file. Your config as mine:
watch section
`gulp.watch('app/pug/**/*.pug', ['pug']);`
pug section
```
gulp.task('pug', function(){
    return gulp.src('app/pug/**/*.pug')
    .pipe(pug({pretty: true}))
    .pipe(gulp.dest('app'))
});
```
Tried to add render by .on method to main gulp method, but this give me nothing (:
```
.on('change', function(event) {
  console.log('File ' + event.path + ' was ' + event.type + ', running tasks...');
});
```
Why I upped this question? because SASS could decide this by
`.on('error', sass.logError))`

**Comment by strugee** [comment]
@Scorpovi4 this is a Gulp problem, not a Pug problem. The fact that Pug throws upon encountering invalid syntax is a perfectly reasonable thing for a JS library to do. You should use [gulp-plumber][1] to fix this; Gulp 4 will also include much better error-handling out of the box, so this won't be a huge problem at that time.

I'll also note that while I think this issue should be closed as a WONTFIX in any case, it's filed under the wrong repository, anyway. See pugjs/gulp-pug

 [1]: https://www.npmjs.com/package/gulp-plumber
